lib/services/authentication.go

```diff
@@ -526,14 +507,10 @@ var authPreferenceMarshaler AuthPreferenceMarshaler = &teleportAuthPreferenceMar
 
 // SetAuthPreferenceMarshaler sets global AuthPreferenceMarshaler
 func SetAuthPreferenceMarshaler(m AuthPreferenceMarshaler) {
-	marshalerMutex.Lock()
-	defer marshalerMutex.Unlock()
 	authPreferenceMarshaler = m
```

Comment:
Why this change - is it safe to assume the marshaller is only set from a single goroutine?

Reply:
The marshalerMutex is made irrelevant by #5384. I'm going to rebase off of #5384 and draft this PR for now.